pyproject.toml

```diff
@@ -2,6 +2,12 @@
 requires = [
     "setuptools >= 40.6.2",
     "wheel >= 0.30.0",
+    "typing_extensions>=3.10",
```

Comment:
Do we still need https://github.com/python/mypy/blob/master/mypy-requirements.txt and https://github.com/python/mypy/blob/master/build-requirements.txt ?

Is there any way to dedup deps? 🤔

Reply:
Unfortunately I'm not really sure of a way to dedup these. We could of course switch to only using pyproject.toml... As long as we are using a PEP 517 build tool it should work fine.

For mypy-requirements.txt, I believe that is specifically for doing the self-check, so somewhat orthogonal to this build requires file... but it might be worth it to add comments to keep them in sync.

Reply:
Yeah, I think adding comments in both places to keep things in sync is a good idea, want to make that change?

Once this is merged, we can get rid of build-requirements.txt.
You'd just need to remove https://github.com/mypyc/mypy_mypyc-wheels/blob/030c76827c6d09dfa5113567ef3405b40679d9f2/build_wheel.py#L49 and all mentions of `PIP_NO_BUILD_ISOLATION=no` in that file

Reply:
Ah, looks like cibuildwheel doesn't support passing config settings yet, so might need to wait for https://github.com/pypa/cibuildwheel/issues/1227

Reply:
In the wheels repo we don't use `--use-mypyc`, we just set `MYPY_USE_MYPYC=1`... are isolated builds clever enough to hide env vars? Or I guess, does pip's impl of them hide env vars?

Reply:
For linux builds cibuildwheel requires you ask for the environment variables to be passed or you can set them manually (because it's running inside a container). But it's easy to do.

Reply:
Yup, we already set `CIBW_ENVIRONMENT`